### ISSUE TITLE
[Bug]: Missing return for non-admin users when renaming objects

### ISSUE THREAD

**Issue by QWpreischl** [pimcore/pimcore#12948]
### Expected behavior

- Non-Admin user renames object in backend
- Object gets renamed

### Actual behavior

- Non-Admin user renames object in backend
- Error message pops up "There was an error while renaming the item." 
- After closing the error message you can see that the renaming on the object worked

### Steps to reproduce

- Login as Non-Admin user
- Rename any object

What fixed the problem for me on Version 10.5.2, but also is still missing in 11.x:
In file:
https://github.com/pimcore/pimcore/tree/11.x/bundles/AdminBundle/Controller/Admin/DataObject/DataObjectController.php
on Line 1101 there is a "return" missing before the adminJson function so the "$success = true" of the renameObject function is passed through to the response of the request:

```
} elseif ($key && $object->isAllowed('rename')) {
            return $this->adminJson($this->renameObject($object, $key));
        } else {
```

**Comment by blankse** [comment]
@QWpreischl I created a PR: https://github.com/pimcore/pimcore/pull/12949

**Comment by dvesh3** [comment]
Fixed by #12949
